tempest2/src/main/kotlin/app/cash/tempest2/internal/DynamoDbScannable.kt

```diff
@@ -66,13 +68,14 @@ internal class DynamoDbScannable<K : Any, I : Any, R : Any>(
       pageSize: Int,
       consistentRead: Boolean,
       filterExpression: Expression?,
-      initialOffset: Offset<K>?
+      initialOffset: Offset<K>?,
+      workerId: WorkerId?
```

Comment:
I don't think I like the idea of having a parallel scanAll. The usecase for `scanAll` is that you expect a single worker to get through the whole table, while the usecase for parallel scan is that you expect the opposite.

I'm open to convincing, but I'm leaning against this currently,

Reply:
I was on the fence there and ended up aiming for symmetry as `WorkerId` lives within `ScanConfig`, but I agree with you. What I'm thinking I could do is to remove `WorkerId` from `scanAll` and add a comment to the overloaded method that takes `ScanConfig` mentioning that `WorkerId `, if provided, will be ignored.